sympy/core/mul.py

```diff
@@ -402,14 +402,30 @@ def _gather(c_powers):
 
         #  0             1
         # x  -> 1       x  -> x
-        for b, e in c_powers:
-            if e is S.One:
-                if b.is_Number:
-                    coeff *= b
-                else:
-                    c_part.append(b)
-            elif e is not S.Zero:
-                c_part.append(Pow(b, e))
+        for i in range(2):
+            ebases = []
+            changed = False
+            for p, e in c_powers:
+                if e.is_zero:
+                    continue
+                if e is S.One:
+                    if p.is_Number:
+                        coeff *= p
+                        continue
+                b = bi = p
+                p = Pow(p, e)
+                if p.is_Pow:
+                    b, e = p.as_base_exp()
+                    if b != bi:
```

Comment:
add 'and not bi.is_Pow` to avoid unevaluated powers (and hopefully fix the single failing test)